tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -4215,15 +4306,15 @@ private[optimizer] object OptimizerCore {
       RefinedType(base, isExact, isNullable, AllocationSite.Anonymous)
 
     def apply(tpe: Type): RefinedType = tpe match {
-      case IntType | FloatType | DoubleType =>
-        RefinedType(tpe, isExact = false, isNullable = false)
-      case BooleanType | StringType | UndefType | NothingType |
-          _:RecordType | NoType =>
-        RefinedType(tpe, isExact = true, isNullable = false)
+      case AnyType | ClassType(_) | ArrayType(_) =>
+        RefinedType(tpe, isExact = false, isNullable = true)
       case NullType =>
         RefinedType(tpe, isExact = true, isNullable = true)
-      case _ =>
-        RefinedType(tpe, isExact = false, isNullable = true)
+      case NothingType | UndefType | BooleanType | CharType | ByteType |
+          LongType | StringType | NoType =>
+        RefinedType(tpe, isExact = true, isNullable = false)
+      case ShortType | IntType | FloatType | DoubleType | RecordType(_) =>
+        RefinedType(tpe, isExact = false, isNullable = false)
```

Comment:
I don't fully understand this. If there are not subtyping relationships between primitive types, why are some of them not exact?